include/fuse_rl/common/sensor_proc.h

```diff
@@ -128,38 +132,46 @@ namespace common
 {
 
 /**
- * @brief Method to create sub-measurements from full measurements and append them to existing partial measurements
+ * @brief Method to merge two vectors of indices adding an offset to the RHS one.
  *
- * base_index and offset are separate parameters to prevent issues with negative size_t values being passed in by users
+ * @param[in] lhs_indices - LHS vector of indices
+ * @param[in] rhs_indices - RHS vector of indices
+ * @param[in] rhs_offset - RHS offset to be added to the RHS vector indices (defaults to 0)
+ */
+inline std::vector<size_t> mergeIndices(const std::vector<size_t>& lhs_indices, const std::vector<size_t>& rhs_indices,
+                                        const size_t rhs_offset = 0u)
+{
+  auto merged_indices = boost::copy_range<std::vector<size_t>>(boost::join(lhs_indices, rhs_indices));
+
+  const auto rhs_it = merged_indices.begin() + lhs_indices.size();
+  std::transform(rhs_it, merged_indices.end(), rhs_it, std::bind2nd(std::plus<size_t>(), rhs_offset));
```

Comment:
`std::bind2nd` is marked as deprecated in C++11.
https://en.cppreference.com/w/cpp/utility/functional/bind12

```suggestion
  std::transform(rhs_it, merged_indices.end(), rhs_it, std::bind(std::plus<size_t>(), std::placeholders::_1, rhs_offset));
```